strongbox-web-core/src/main/java/org/carlspring/strongbox/services/DirectoryListingServiceImpl.java

```diff
@@ -131,82 +163,86 @@ private DirectoryListing fromPath(Path path)
 
             file.setStorageId((String) fileAttributes.get(RepositoryFileAttributeType.STORAGE_ID.getName()));
             file.setRepositoryId((String) fileAttributes.get(RepositoryFileAttributeType.REPOSITORY_ID.getName()));
+            file.setArtifactPath((String) fileAttributes.get(RepositoryFileAttributeType.ARTIFACT_PATH.getName()));
+            file.setLastModified(new Date(((FileTime) fileAttributes.get("lastModifiedTime")).toMillis()));
+            file.setSize((Long) fileAttributes.get("size"));
 
-            file.setArtifactPath((String) fileAttributes.get("artifactPath"));
+            // Depending on scanned file/path this might not be a `storage` artifact (i.e. it could be a log file)
+            // in which case we need to set the artifactPath to the file's path.
+            if (file.getArtifactPath() == null)
+            {
+                file.setArtifactPath(sanitizedPath(relativePath.toString() + "/" + file.getName()));
+            }
 
+            UriComponentsBuilder builder = null;
             if (Boolean.TRUE.equals(fileAttributes.get("isDirectory")))
             {
-                file.setUrl(calculateDirectoryUrl(file));
-
-                directories.add(file);
-
-                continue;
+                // TODO: Use `UriComponentsBuilder.clone()` when spring-framework/issues/24772 is fixed and released
```

Comment:
Could you please create a follow-up task for this and link to the Spring issue, and make the new issue depend on this pull request? Thanks!